src/shared/helpers/getActualDataList.ts

```diff
@@ -1,30 +1,54 @@
 import { getCourseDate } from './getCourseDate';
 import { dayJS } from '@/app/services/dayjs';
+import { Course } from '@/app/types';
+import { isCourse } from '@/entities/courses/helpers/is-course';
+import { EventCardProps } from '@/entities/events';
 
-type DataListType<T> = Array<T>;
-type DataItem = Record<string, string | HTMLElement>;
-interface DataItemWithDate extends DataItem {
-  date: string; // 'YYYY-MM-DD'
-}
+type DataListType = (Course | EventCardProps)[];
 
-export type ActualDataListProps = {
-  dataList: DataListType<DataItemWithDate>;
+export type ActualDataListParams = {
+  dataList: DataListType;
   actualDelayInDays: number;
-  sorted?: boolean;
+  filtered?: boolean;
 };
 
-export const getActualDataList = ({
+type getActualDataListType = (props: ActualDataListParams) => DataListType;
+
+const filterDataList = (dataList: DataListType) =>
+  dataList.filter((item) => {
+    const date = isCourse(item) ? item.startDate : item.date;
+
+    return date !== 'TBD';
```

Comment:
We don't use `TBD` in `courses.data.ts`, so this line doesn't make any sense to me

Reply:
This func only used in getDataListType, which one first mapped irrelevant datas as TBD, an then filters by it